client/my-sites/controller.js

```diff
@@ -137,7 +157,7 @@ function onSelectedSiteAvailable( context ) {
 
 	if ( isDomainOnlySite( state, selectedSite.ID ) &&
 		! isPathAllowedForDomainOnlySite( context.pathname, selectedSite.slug ) ) {
```

Comment:
I added a commit to change this to a canonical whitelist instead of a whitelist of prefixes.